src/persist-types/src/part.rs

```diff
@@ -223,72 +224,99 @@ impl Part {
 /// An in-progress columnar constructor for one blob's worth of data.
 #[derive(Debug)]
 pub struct PartBuilder {
-    key: DynStructMut,
-    val: DynStructMut,
-    ts: Vec<i64>,
-    diff: Vec<i64>,
+    /// The key column.
+    pub key: ColumnsMut,
+    /// The val column.
+    pub val: ColumnsMut,
+    /// The ts column.
+    pub ts: Codec64Mut,
+    /// The diff column.
+    pub diff: Codec64Mut,
 }
 
 impl PartBuilder {
-    /// Returns a new PartBuilder with the given schema.
-    pub fn new<K, KS: Schema<K>, V, VS: Schema<V>>(key_schema: &KS, val_schema: &VS) -> Self {
-        let key = ColumnMut::<DynStructCfg>::new(&key_schema.columns());
-        let val = ColumnMut::<DynStructCfg>::new(&val_schema.columns());
-        let ts = Vec::new();
-        let diff = Vec::new();
-        PartBuilder { key, val, ts, diff }
-    }
+    /// Returns a new [`PartBuilder`] and [`PartBuilderSeed`] with the given schema.
+    ///
+    /// See [`PartBuilderSeed::grow`] for how to finalize into a [`Part`].
+    pub fn new<K, KS: Schema<K>, V, VS: Schema<V>>(
```

Comment:
Naive question: I don't immediately see why the seed gets passed around separately instead of getting passed through as an internal field of PartBuilder?